File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/usecase/ProcessScribbleUseCase.kt
package com.scribblefit.feature.ai.domain.usecase

import com.scribblefit.feature.ai.domain.repository.CanvasRepository

/**
 * Orchestrates adding a raw user entry to the feed and triggering background parsing.
 */
class ProcessScribbleUseCase(
    private val canvasRepository: CanvasRepository
) {
    suspend operator fun invoke(rawText: String) {
        canvasRepository.addScribble(rawText)
    }
}

File: apps/ios/ScribbleFit/ScribbleFit/Features/AI/Domain/UseCases/ProcessScribbleUseCase.swift
import Foundation

/**
 * Orchestrates adding a raw user entry to the feed and triggering background parsing.
 */
public final class ProcessScribbleUseCase {
    private let canvasRepository: CanvasRepository
    
    public init(canvasRepository: CanvasRepository) {
        self.canvasRepository = canvasRepository
    }
    
    public func execute(rawText: String) async throws {
        try await canvasRepository.addScribble(rawText: rawText)
    }
}
